### PR TITLE
Annotate Min and Max WorkingSet setters as supported on macOS and FreeBSD

### LINKED ISSUE #41595: MaxWorkingSet supported platforms
https://github.com/dotnet/runtime/blob/3130ac980c97c0664eddb04411f8f765a9f908d8/src/libraries/System.Diagnostics.Process/ref/System.Diagnostics.Process.cs#L45

The code specifies supported platforms as Windows only.

However, the [documentation](https://github.com/dotnet/docs/blob/3588fdadac34bee137fe519d50c9a187d704d1fd/docs/core/compatibility/unsupported-apis.md) mentions it throws in Linux only, which means both Windows and macOS are supported.

I think macOS should be missing in either the documentation or the supported platforms attribute.

### PR DESCRIPTION
**Pull request by adamsitnik**
Fixes #41595

### CODE REVIEW

File: src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs [hunk 1]
```diff
@@ -254,12 +254,10 @@ public string MachineName
...
         }
 
-        /// <devdoc>
-        ///    <para>
-        ///       Gets or sets the maximum allowable working set for the associated
-        ///       process.
-        ///    </para>
-        /// </devdoc>
+        /// <summary>
+        /// Gets or sets the maximum allowable working set for the associated process.
```
Comment (eiriktsarpalis) [inline src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs:258]: Docs changes need to be ported manually to dotnet-api-docs

**Review by eiriktsarpalis — APPROVED** [review]
LGTM

**Review by jeffhandley — APPROVED** [review]

**Comment by jeffhandley** [comment]
@adamsitnik / @eiriktsarpalis As soon as this is green, I'll merge it in, create the backport PR for RC2, and send it to Tactics for approval.

**Comment by jeffhandley** [comment]
The checks are timing out.  I'm going to go ahead and create the backport to get it going.

/backport to release/5.0-rc2

**Comment by jeffhandley** [comment]
Check failure and timeouts are unrelated.

**Pull request merged**